fix: Remove unused @types/node dependencies

tylerbutler opened the pull request:
Remove unused @types/node dependencies.

[AB#3098](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3098)

tylerbutler committed:
fix: Remove unused @types/node dependencies

tylerbutler committed:
lockfiles

curtisman commented:
I think that here are additional work to clean up all the "node" references, beyond the dependencies.

e.g.  [this](https://github.com/microsoft/FluidFramework/blob/c2b8d3c31f51764834bfa2655c92b2abf67083fe/packages/common/container-definitions/tsconfig.json#L10) will still introduce the `/// <reference type=node" />` in the .d.t.s

curtisman commented:
I am not 100% sure but remove the dependency without removing the injected reference type line in `.d.ts` will cause compilation errors with project that doesn't have `@types/node` dependency already?

tylerbutler committed:
remove type references

tylerbutler committed:
types

tylerbutler committed:
quorum

tylerbutler committed:
Merge branch 'main' into node-types-deps

tylerbutler committed:
Merge branch 'main' into node-types-deps

tylerbutler committed:
fixes

tylerbutler committed:
Merge branch 'main' into node-types-deps

tylerbutler commented:
@curtisman You were right; when I removed the types from tsconfig for container-runtime and container-definitions, several downstream packages broke because of missing types. I think I fixed everything up. CI passes, at least.

tylerbutler committed:
Merge branch 'main' into node-types-deps

curtisman commented on experimental/PropertyDDS/packages/property-dds/tsconfig.json:
```diff
@@ -8,7 +8,8 @@
 		"outDir": "./dist",
-		"rootDir": "./src"
+		"rootDir": "./src",
+        "types": ["node"]
```
this is interesting that property-dds needs node types?  Probably no harm for add it for now.

tylerbutler replied:
property-dds uses `buffer` and `stream`.

tylerbutler committed:
formatting

tylerbutler committed:
Merge branch 'main' into node-types-deps
